test/unit/SourceLocator/SourceStubber/ReflectionSourceStubberTest.php

```diff
@@ -307,30 +306,19 @@ private function assertSameParameterAttributes(
             . '.' . $original->getName();
 
         self::assertSame($original->getName(), $stubbed->getName(), $parameterName);
-        self::assertSame($original->isArray(), $stubbed->isArray(), $parameterName);
-        if (! ($original->getDeclaringClass()->getName() === Closure::class && $originalMethod->getName() === 'fromCallable')) {
-            // Bug in PHP: https://3v4l.org/EeHXS
-            self::assertSame($original->isCallable(), $stubbed->isCallable(), $parameterName);
-        }
+        self::assertSame(@$original->isArray(), $stubbed->isArray(), $parameterName);
+        self::assertSame(@$original->isCallable(), $stubbed->isCallable(), $parameterName);
 
         //self::assertSame($original->allowsNull(), $stubbed->allowsNull()); @TODO WTF?
-        if ($original->getDeclaringClass()->getName() !== 'FFI') {
-            // Parameters can be passed by reference and also by value
-            self::assertSame($original->canBePassedByValue(), $stubbed->canBePassedByValue(), $parameterName);
-        }
 
-        if (
-            ! in_array($parameterName, ['mysqli_stmt#bind_param.vars', 'mysqli_stmt#bind_result.vars'], true)
-            && ! preg_match('~^RedisCluster#\w+.arg$~', $parameterName)
-        ) {
-            // Parameters are variadic but not optional
-            self::assertSame($original->isOptional(), $stubbed->isOptional(), $parameterName);
-        }
+        self::assertSame($original->canBePassedByValue(), $stubbed->canBePassedByValue(), $parameterName);
+
+        self::assertSame($original->isOptional(), $stubbed->isOptional(), $parameterName);
 
         self::assertSame($original->isPassedByReference(), $stubbed->isPassedByReference(), $parameterName);
         self::assertSame($original->isVariadic(), $stubbed->isVariadic(), $parameterName);
 
-        $class = $original->getClass();
+        $class = @$original->getClass();
```

Comment:
is `getClass()` deprecated too?

Reply:
Yes https://www.php.net/manual/en/reflectionparameter.getclass.php